qiskit/ignis/experiments/calibration/generators/single_qubit.py

```diff
@@ -0,0 +1,200 @@
+    of that pulse. Note that the same pulse may be applied to multiple qubits
+    to perform simultaneous calibration.
+    """
+    def __init__(self,
+                 name: str,
+                 qubit: int,
+                 parameters: Dict[str, Union[int, float, complex, circuit.Parameter]],
+                 values_to_scan: Iterable[float],
+                 ref_frequency: Optional[Union[float, circuit.Parameter]] = None,
+                 pulse_envelope: Optional[Callable] = None):
+        """
+        Args:
+            name: Name of this experiment.
+            qubit: The qubit to which we will add the calibration.
+            parameters: The arguments for the pulse schedule.
+            values_to_scan: A list of parameter values for which to generate a circuit.
+            ref_frequency: A reference frequency of drive channel to run calibration.
+                Usually this is identical to the qubit frequency, or f_01.
+                The drive channel is initialized with this frequency and
+                shift frequency instruction of consecutive pulses, or pulse sideband,
+                are modulated with respect to this frequency.
+            pulse_envelope: A parametric pulse function used to generate the schedule
+                that is added to the circuits. This defaults to Gaussian and
+                parameters should contain `duration`, `amp`, and `sigma`.
+        """
+        self._pulse_envelope = pulse_envelope
+
+        super().__init__(name=name,
+                         qubit=qubit,
+                         parameters=parameters,
+                         values_to_scan=values_to_scan,
+                         ref_frequency=ref_frequency)
+
+    def _template_qcs(self) -> List[circuit.QuantumCircuit]:
+        """Create the template quantum circuit.
+        """
+        cal_sched = self.single_pulse_schedule()
+        parameter = list(cal_sched.parameters)[0]
+
+        template_qc = circuit.QuantumCircuit(1, 1, name='null')
+        gate = circuit.Gate(parameter.name, 1, [parameter])
+        template_qc.append(gate, [0])
+        template_qc.add_calibration(gate, self.qubits, cal_sched, parameter)
+        template_qc.measure(0, 0)
+
+        return [template_qc]
+
+    def single_pulse_schedule(self) -> pulse.Schedule:
+        """
+        Creates the schedules that will be added to the circuit.
+        """
+        with pulse.build() as sched:
+            d_channel = pulse.DriveChannel(self.qubits[0])
+            # initialize channel frequency
+            if self._ref_frequency is not None:
+                pulse.set_frequency(self._ref_frequency, d_channel)
+            # add single stimulus pulse
+            if not self._pulse_envelope:
+                # remove DRAG coefficient if exists
+                self._parameters.pop('beta', None)
+                pulse.play(pulse.Gaussian(**self._parameters), d_channel)
```

Comment:
Perhaps we should default to Drag?

Reply:
what happens if we run spectroscopy with junk beta?

Reply:
ah, ok, it doesn't impact to frequency. yeah, perhaps we can replace this with Darg.